src/DynamoUtilities/PathHelper.cs

```diff
@@ -348,5 +350,59 @@ internal static string ValidateDirectory(string directoryPath, bool absolutePath
 
             return directoryPath;
         }
+
+        internal class DirectoryPathComparer : IEqualityComparer<string>
```

Comment:
are there any constraints on the paths, for example, should they always be absolute or can relative paths be safely checked?

Reply:
Well...relative paths could be used with this class...I did not write it so it could only work for the trust APIs...
Ex:
AbsolutPath : C:/a/b/c/
RelativePAth: b/c/ (with current dir C:/a) 
These paths would be found equal by the comparer

Reply:
Removed this class since there were modifications to the trusted API (by morpheus team) that made this obsolete